src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,41 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        User user = new User();
+
+        Optional<User> userByEmail = userService.findByEmail(email);
+        if (userByEmail.isEmpty()) {
+            user.setEmail(email);
+            user.setPassword(password);
+            return user;
+        } else {
+            throw new AuthenticationException("User with email " + email + " already exist");
+        }
+    }
```

Comment:
The `login` method logic is incorrect. It should authenticate an existing user by checking if the email exists and then verifying the password. Currently, it throws an exception if the user exists, which is not the expected behavior for a login method.